.github/workflows/sanity-test.yaml

```diff
@@ -0,0 +1,244 @@
+          then
+              echo "repo could not be connected"
+              exit 1
+          fi
+
+      # run the tests
+      - name: Backup exchange test
+        id: exchange-test
+        run: |
+          ./corso backup create exchange --user "${CORSO_M365_TEST_USER_ID}" --hide-progress 2>&1 | tee $TEST_RESULT/backup_exchange.txt 
+          valid_num=$(grep -i -e 'Completed (0 errors)' $TEST_RESULT/backup_exchange.txt | wc -l)
+          if [ $valid_num -ne 1 ]
+          then
+              echo "backup was not successfull"
+              exit 1
+          fi
+          echo result=$(grep -i -e 'Completed (0 errors)' $TEST_RESULT/backup_exchange.txt  | awk '{print $2}') >> $GITHUB_OUTPUT
+
+      # list the backup exhange
+      - name: Backup exchange list test
+        run: |
+          set -euo pipefail
+          touch $TEST_RESULT/backup_exchange_list.txt
+          ./corso backup list exchange --hide-progress 2>&1 | tee $TEST_RESULT/backup_exchange_list.txt
+          cat $TEST_RESULT/backup_exchange_list.txt 
+          valid_num=$(grep -i -e ${{ steps.exchange-test.outputs.result }} $TEST_RESULT/backup_exchange_list.txt | wc -l)
+         
+          if [ $valid_num -ne 1 ]
+          then
+              echo "listing of backup was not successfull"
+              exit 1
+          fi
+
+      # AWS creds
+      - name: Configure AWS credentials from Test account
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.AWS_IAM_ROLE }}
+          role-session-name: integration-testing
+          aws-region: us-east-1
+
+      # test incremental backup exhange
+      - name: Backup exchange incremental
+        id: exchange-incremental-test
+# TODO: add backup time check
+        run: |
+          set -euo pipefail
+          ./corso backup create exchange --user "${CORSO_M365_TEST_USER_ID}" --data 'email' --hide-progress 2>&1 | tee $TEST_RESULT/backup_exchange_incremental.txt 
+          valid_num=$(grep -i -e 'Completed (0 errors)' $TEST_RESULT/backup_exchange_incremental.txt | wc -l)
+
+          if [ $valid_num -ne 1 ]
+          then
+              echo "backup was not successful"
+              exit 1
+          fi
+
+          echo result=$(grep -i -e 'Completed (0 errors)' $TEST_RESULT/backup_exchange_incremental.txt  | awk '{print $2}') >> $GITHUB_OUTPUT
+
+      # test exchange restore
+      - name: Backup exchange restore
+        id: exchange-restore-test
+        run: |
+          set -euo pipefail
+          ./corso restore exchange --backup "${{ steps.exchange-incremental-test.outputs.result }}" 2>&1 | tee $TEST_RESULT/exchange-restore-test.txt
+          echo result=$(grep -i -e 'Restoring to::' $TEST_RESULT/exchange-restore-test.txt | awk -F "::" '{print $2}' )  >> $GITHUB_OUTPUT
+        
+        
+      - name: Restoration check
+        env:
+          RESTORE_FOLDER:  ${{ steps.exchange-restore-test.outputs.result }}
+          EXCHANGE_TEST: "true"
```

Comment:
This should probably be something like `RESTORE_SERVICE: "exchange"` as we will have more services in future. Same change on the go side.